### PR TITLE
pdksync - (PDKSync) Removal of puppet_module_gems

### PR DESCRIPTION
**Pull request by david22swan**
pdksync/remove_puppet_module_gems
pdk version: `2.5.0`

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -6,3 +6,4 @@ RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
 # [Optional] Uncomment this section to install additional packages.
 # RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
 #     && apt-get -y install --no-install-recommends <your-package-list-here>
+
```
Comment (bastelfreak) [inline .devcontainer/Dockerfile:9]: is that newline on purpose?

Comment (bastelfreak) [reply .devcontainer/Dockerfile:9]: I noticed you set this to unmanaged. Should the file be deleted?

File: Gemfile [hunk 2]
```diff
@@ -13,25 +13,31 @@ def location_for(place_or_version, fake_version = nil)
...
-  gem "json", '= 2.3.0',                                         require: false if Gem::Requirement.create(['>= 2.7.0', '< 2.8.0']).satisfied_by?(Gem::Version.new(RUBY_VERSION.dup))
-  gem "puppet-module-posix-default-r#{minor_version}", '~> 1.0', require: false, platforms: [:ruby]
-  gem "puppet-module-posix-dev-r#{minor_version}", '~> 1.0',     require: false, platforms: [:ruby]
-  gem "puppet-module-win-default-r#{minor_version}", '~> 1.0',   require: false, platforms: [:mswin, :mingw, :x64_mingw]
-  gem "puppet-module-win-dev-r#{minor_version}", '~> 1.0',       require: false, platforms: [:mswin, :mingw, :x64_mingw]
-  gem "voxpupuli-puppet-lint-plugins", '>= 3.0',                 require: false, platforms: [:ruby]
-  gem "ruby-pwsh",                                               require: false
-  gem "github_changelog_generator",                              require: false
+  gem "json", '~> 2.0',                                require: false
+  gem "voxpupuli-puppet-lint-plugins", '~> 3.0',       require: false
```
Comment (bastelfreak) [inline Gemfile:18]: 3.0 pulled in some older version of puppet-lint plugins that could lead to wrong autofixes. We released 3.1 with a fix: https://github.com/voxpupuli/voxpupuli-puppet-lint-plugins/pull/26

Comment (david22swan) [reply Gemfile]: updated

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (david22swan) [reply .devcontainer/Dockerfile]: Ah no, that was so I could add the apt-get update

**Review by LukasAud — APPROVED** [review]
LGTM

**Pull request merged**